### PR TITLE
repositoryの`UpsertGameCreatorPresetJobsRelations`実装

### PR DESCRIPTION
**Pull request by ikura-hamu**
fix #1526

- **:white_check_mark: TestUpsertGameCreatorPresetJobsRelationsの仮実装**
- **:sparkles: UpsertGameCreatorPresetJobsRelationsの実装**
- **:white_check_mark: TestUpsertGameCreatorPresetJobsRelationsのテストケース実装**
- **:memo: interface にコメント追加**

## Summary by CodeRabbit

* **New Features**
  * Implemented batch association functionality for linking game creators with preset jobs
  * Enhanced error handling to prevent invalid associations with non-existent creators or jobs

* **Tests**
  * Added comprehensive test coverage validating creator-job association behavior including edge cases and error scenarios

### CODE REVIEW

File: src/repository/gorm2/game_creator_test.go [hunk 1]
```diff
@@ -664,3 +665,199 @@ func TestCreateGameCreators(t *testing.T) {
...
+					require.NoError(t, err)
+					err = db.Unscoped().Model(testCase.afterCreatorPresetJobsRelations).
+						Association("CreatorJobs").Clear()
+					require.NoError(t, err)
+					err = db.Unscoped().Session(&gorm.Session{AllowGlobalUpdate: true}).
+						Delete(testCase.afterCreatorPresetJobsRelations).Error
+					require.NoError(t, err)
+					err = db.Model(&schema.GameTable2{}).Unscoped().Session(&gorm.Session{AllowGlobalUpdate: true}).
+						Delete(testCase.afterCreatorPresetJobsRelations).Error
+					require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (ikura-hamu) [reply src/repository/gorm2/game_creator_test.go:1150]: 実際に発行されてるクエリを見ると、game の id を指定して削除しているので問題ない。

File: src/repository/gorm2/game_creator_test.go [hunk 2]
```diff
@@ -991,3 +992,199 @@ func TestGetGameCreatorsByUserIDs(t *testing.T) {
...
+
+			assert.Len(t, result, len(testCase.afterCreatorPresetJobsRelations))
+			for i, expected := range testCase.afterCreatorPresetJobsRelations {
+				actual := result[i]
+				assert.Equal(t, expected.ID, actual.ID)
+				assert.Equal(t, expected.UserID, actual.UserID)
+				assert.Equal(t, expected.UserName, actual.UserName)
+				assert.WithinDuration(t, expected.CreatedAt, actual.CreatedAt, time.Second)
+
+				assert.Len(t, actual.CreatorJobs, len(expected.CreatorJobs))
```
Comment (mathsuky) [inline src/repository/gorm2/game_creator_test.go:1177]: ここでもし不正な長さであった場合，後続の`actualJob := actual.CreatorJobs[j]`がpanicして結局その他のテストが実行されないまま終了するので，`require`を用いても良いと思いました。

File: src/repository/gorm2/game_creator_test.go [hunk 3]
```diff
@@ -991,3 +992,199 @@ func TestGetGameCreatorsByUserIDs(t *testing.T) {
...
+				return
+			}
+
+			assert.NoError(t, err)
+
+			var result []schema.GameCreatorTable
+			err = db.Preload("Game").Preload("CreatorJobs").Order("created_at ASC").Find(&result).Error
+			assert.NoError(t, err)
+
+			assert.Len(t, result, len(testCase.afterCreatorPresetJobsRelations))
```
Comment (mathsuky) [inline src/repository/gorm2/game_creator_test.go:1169]: ここでもし不正な長さであった場合，後続の`actual := result[i]`がpanicして結局その他のテストが実行されないまま終了するので，`require`を用いても良いと思いました。

**Review by mathsuky — COMMENTED** [review]
レビュー遅くなりすみません🙇
テスト，動作ともに全く問題ないと思います。ありがとうございます。
軽微なコメントを付したので簡単にご確認いただけますと幸いです。
